### ISSUE TITLE
missing data imputation for linear regression

### ISSUE THREAD

**Issue by murphyk** [probml/pyprobml#460]
In [sec 15.2.2 of Statistical Rethinking (2nd end)](https://github.com/Booleans/statistical-rethinking/blob/master/Statistical%20Rethinking%202nd%20Edition.pdf), there is a nice example of fitting a multivariate Gaussian to the inputs of a linear regression model, in order to impute missing values. These are then used to predict an outcome. Some numpyro code for this is [in code snippet 15.22 here](https://fehiepsi.github.io/rethinking-numpyro/15-missing-data-and-other-opportunities.html). 

The goal of this task is as follows:
1. rewrite the code in a more general way, so it works with D-dimensional input vectors x (not just 2 dimensional mass and brain size)
2.  apply it to a more "ML relevant" example, such as the [boston housing dataset](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_boston.html#sklearn.datasets.load_boston). The inputs can be  made missing artificially so we can control the difficulty of the problem, and compare to the fully observed scenario.
3. Compute predictive accuracy as a function of the amount of missing data, for a naive model that drops missing cases, and a simple baseline that does mean value imputation, and the full Bayesian model described above.

**Comment by always-newbie161** [comment]
working on this

**Comment by murphyk** [comment]
I moved this to hermes
